packages/language-server/src/server.ts

```diff
@@ -63,6 +64,22 @@ connection.onCompletionResolve(
   }
 );
 
+documents.onDidChangeContent(async changeEvent => {
+  // TODO: can we ensure the promise has always been initialized first (never be undefined)?
+  if (getSemanticModelPromise === undefined) {
+    return;
+  }
+  // TODO: how to debounce this? should we debounce this?
```

Comment:
I don't understand what you mean with this comment

Reply:
- How many such events happen per second? 
- Does the changeEvent only gets triggered after N miliseconds after there has been no input?
- Does the changeEvent happen for every character cicked?
- Do we need to optimize this or does not LSP implementation does it?

Reply:
I am going to assume the Node LSP library handles this, otherwise everyone would have major performance issues.